lexers/org.lua

```diff
@@ -0,0 +1,137 @@
+-- Also, I don’t think that the TAG is the right name, just use a custom tag
+-- local priority = lex:tag('PRIORITY', '[#' * S('ABC') * ']')
+-- local tags = lex:tag('TAG', ':' * (lexer.word + lexer.punct)^1 * lexer.newline)
+
+-- XXX not sure what it is good for
+-- Keywords.
+-- lex:add_rule('keyword', lex:tag(lexer.KEYWORD, lex:word_match(lexer.KEYWORD)))
+-- lex:set_word_list(lexer.KEYWORD, {'SCHEDULED', 'DEADLINE', 'CLOSED'})
+
+-- DateTime.
+local DD =   lexer.digit * lexer.digit
+local date = DD * DD * '-' * DD * '-' * DD
+lex:set_word_list('wday', {
+	'Mo', 'Di', 'Mi', 'Do', 'Fr', 'Sa', 'So',
+	'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun',
+	'Po', 'Út', 'St', 'Čt', 'Pá', 'So', 'Ne'
+})
+lex:set_word_list('weekday', {
+	'Montag', 'Dienstag', 'Mittwoch', 'Donnerstag', 'Freitag', 'Samstag', 'Sonntag',
+	'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday',
+	'Pondělí', 'Úterý', 'Středa', 'Čtvrtek', 'Pátek', 'Sobota', 'Neděle'
+})
+lex:set_word_list('month', {
+	'Januar', 'Februar', 'März', 'April', 'Mai', 'Juni', 'Juli', 'August', 'September', 'Oktober', 'November', 'Dezember',
+	'January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December',
+	'Leden', 'Únor', 'Březen', 'Duben', 'Květen', 'Červen', 'Červenec', 'Srpen', 'Září', 'Říjen', 'Listopad', 'Prosince'
+})
+local time_range = (' ' * DD * ':' * DD)^0 * ('-' * DD * ':' * DD)^0
+local repeater =   (' +' * lexer.integer * S('dwmy'))^0
+local alarm =      (' -' * lexer.integer * S('dwmy'))^0
+local pattern_datetime1 = S('<[')^-1 * 'date' * 'wday' * time_range * repeater * alarm * S('>]')^-1
+local pattern_datetime2 = lexer.starts_line('weekday') * lexer.space^1 * DD * '. ' * 'month' * lexer.space^1 * DD * DD
+local datetime =   pattern_datetime1 + pattern_datetime2
+
+lex:add_rule('current_date', lex:tag('CURRENT_DATE', lexer.starts_line(lex:word_match('weekday')) * lexer.space^1 * DD * '. ' * lex:word_match('month') * lexer.space^1 * DD * DD * '|'))
+lex:add_rule('time', lex:tag(lexer.CLASS, DD * ':' * DD))
+lex:add_rule('week', lex:tag('UNDERLINE', lexer.starts_line('KW ' * DD * lexer.space^25) + lexer.starts_line('Wk ' * DD * lexer.space^25)))
+lex:add_rule('datetime', lex:tag(lexer.NUMBER, lex:word_match('datetime')))
+
+-- Heading patterns.
+local pattern_priority = '[#' * S('ABC') * ']'
+local pattern_tags = ':' * (lexer.word + lexer.punct)^1 * lexer.newline
+local pattern_h1 = lexer.starts_line('* ')        -- Heading1.
+local pattern_h2 = lexer.starts_line('** ')       -- Heading2.
+local pattern_h3 = lexer.starts_line('*** ')      -- Heading3.
+local pattern_h4 = lexer.starts_line('**** ')     -- Heading4.
+local pattern_h5 = lexer.starts_line('***** ')    -- Heading5.
+
+-- Heading lex:tag parts.
+local part_h1 = lex:tag(lexer.CLASS,        (lexer.nonnewline - S('<[') - (':' * lexer.word))^0) * datetime^0 * lex:tag(lexer.CLASS,        (lexer.nonnewline - (':' * lexer.word))^0)
```

Comment:
For this and all header-related tags, would you mind using the `lexer.HEADING` tag? For example, Markdown's h1 is `lexer.HEADING .. '.h1'`, h2 is `lexer.HEADING .. '.h2'`, etc.

Tags should be descriptive, not based on other tags that have the desired style/color in the current theme.

Reply:
Whole headings stuff copied from Markdown lexer.